addons/mail/static/src/new/core/thread_model.js

```diff
@@ -0,0 +1,75 @@
+/** @odoo-module */
+
+import { Composer } from "./composer_model";
+import { Partner } from "./partner_model";
+
+export class Thread {
+    static insert(state, data) {
+        if (data.id in state.threads) {
+            return state.threads[data.id];
+        }
```

Comment:
As discussed yesterday insert should also update data if the record already exists, not just do a create or nothing.

Not blocking if this is an intermediary step to move code into the class and the change is planned quickly after.